lib/userAgent.js

```diff
@@ -34,6 +36,11 @@ module.exports = function (userAgentString) {
   this.uaOSVersion = getVersion(userAgentData.os) || null
   this.uaDeviceType = getDeviceType(userAgentData) || null
 
+  if (! this.uaBrowser && ! this.uaOS) {
+    // In the worst case, fall back to a truncated user agent string
+    this.uaBrowser = truncate(userAgentString || '', log)
```

Comment:
Is it useful for use to be able to show a longer (the full) name at some
point? If so, perhaps the truncation could happen in the client, when
displaying that list...

On Wed, Aug 17, 2016, 9:38 AM Vlad Filippov <notifications@github.com>
wrote:

> In lib/userAgent.js
> <https://github.com/mozilla/fxa-auth-server/pull/1422#discussion_r75123628>
> :
>
> > @@ -34,6 +36,11 @@ module.exports = function (userAgentString) {
> >    this.uaOSVersion = getVersion(userAgentData.os) || null
> >    this.uaDeviceType = getDeviceType(userAgentData) || null
> >
> > +  if (! this.uaBrowser && ! this.uaOS) {
> > +    // In the worst case, fall back to a truncated user agent string
> > +    this.uaBrowser = truncate(userAgentString || '', log)
>
> 👍
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mozilla/fxa-auth-server/pull/1422/files/7f7926e037dab27db2c84fef14696b434ca5dc8b#r75123628>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AADJF6HFUNXla9y4qTpQUwHa8MZRMzQ-ks5qgw7XgaJpZM4JmNaw>
> .
>